olp-cpp-sdk-core/src/client/OlpClientSettingsFactory.cpp

```diff
@@ -42,7 +45,11 @@ OlpClientSettingsFactory::CreateDefaultNetworkRequestHandler(
 std::unique_ptr<cache::KeyValueCache>
 OlpClientSettingsFactory::CreateDefaultCache(cache::CacheSettings settings) {
   auto cache = std::make_unique<cache::DefaultCache>(std::move(settings));
-  cache->Open();
+  auto error = cache->Open();
+  if (error == cache::DefaultCache::StorageOpenResult::OpenDiskPathFailure)
+  {
+      OLP_SDK_LOG_ERROR(kLogTag, "Error opening disk cache");
+  }
```

Comment:
Seems you missed two very important points from the ticket:

"In case the DefaultCache fails to open the mutable/protected cache we need to print a _fatal error_ and return _nullptr_ back so that the user knows something bad happened."